internal/pkg/githubapi/pr_metrics.go

```diff
@@ -0,0 +1,108 @@
+package githubapi
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-github/v62/github"
+	lru "github.com/hashicorp/golang-lru/v2"
+	log "github.com/sirupsen/logrus"
+	prom "github.com/wayfair-incubator/telefonistka/internal/pkg/prometheus"
+)
+
+const (
+	timeToDefineStale = 20 * time.Minute
+	metricRefreshTime = 60 * time.Second
+)
+
+func MainGhMetricsLoop(mainGhClientCache *lru.Cache[string, GhClientPair]) {
+	for t := range time.Tick(metricRefreshTime) {
+		log.Debugf("Updating pr metrics at %v", t)
+		getPrMetrics(mainGhClientCache)
+	}
+}
+
+func getRepoPrMetrics(ctx context.Context, ghClient GhClientPair, repo *github.Repository) (prWithStaleChecks int, openPRs int, openPromotionPrs int, err error) {
+	log.Debugf("Checking repo %s", repo.GetName())
+	ghOwner := repo.GetOwner().GetLogin()
+	prListOpts := &github.PullRequestListOptions{
+		State: "open",
+	}
+	prs := []*github.PullRequest{}
+
+	// paginate through PRs, there might be lots of them.
```

Comment:
Maybe we can consider concurrently processing and exposing the metrics if there might be lots of PR.
But i think for now it's okay.

Reply:
We are not in any particular "rush" to finish this, unless it times out.
If we reach a point when it times out **and it doesn't make sense to increase the duration** we change it